src/OrleansRuntime/Catalog/GrainCreator.cs

```diff
@@ -60,25 +63,38 @@ public Grain CreateGrainInstance(Type grainType, IGrainIdentity identity)
         /// <param name="grainType"></param>
         /// <param name="identity">Identity for the new grain</param>
         /// <param name="stateType">If the grain is a stateful grain, the type of the state it persists.</param>
-        /// <param name="storageProvider">If the grain is a stateful grain, the storage provider used to persist the state.</param>
+        /// <param name="persistenceProvider">If the grain is a stateful grain, the provider used to persist the state.</param>
         /// <returns></returns>
-        public Grain CreateGrainInstance(Type grainType, IGrainIdentity identity, Type stateType,
-            IStorageProvider storageProvider)
+        public Grain CreateGrainInstance(Type grainType, IGrainIdentity identity, Type stateType, IPersistenceProvider persistenceProvider)
         {
             //Create a new instance of the grain
             var grain = CreateGrainInstance(grainType, identity);
 
             var statefulGrain = grain as IStatefulGrain;
+            var logViewGrain = grain as ILogViewGrain;
 
-            if (statefulGrain == null)
-                return grain;
+            if (statefulGrain != null)
+            {
+                var storage = new GrainStateStorageBridge(grainType.FullName, statefulGrain, (IStorageProvider) persistenceProvider);
```

Comment:
If IStatefulGrain and ILogViewGrain have different initializations, then I suggest providing different create calls or initialization logic.  Special casing this logic for each grain type is a recipe for spaghetti code.